java-build/modules/common/added/utils/start.sh

```diff
@@ -88,29 +88,35 @@ function exit_flag {
 }
 
 function get_app_file {
-    # For JAR based applications (APP_MAIN_CLASS set), look for a single JAR file if APP_FILE
-    # is not set and use that. If there is not exactly 1 jar APP_FILE will remain unset.
-    # For Python applications, look for a single .py file
-    local cnt
-    if [ -z "$APP_FILE" ]; then
-        if [ -n "$APP_MAIN_CLASS" ]; then
-            cnt=$(cd $APP_ROOT/src/; ls -1 *.jar | wc -l)
-            if [ "$cnt" -eq "1" ]; then
-                APP_FILE=$(cd $APP_ROOT/src/; ls *.jar)
-            else
-                echo "Error, no APP_FILE set and $cnt JAR file(s) found"
-                app_exit
-            fi
-        else
-            cnt=$(cd $APP_ROOT/src/; ls -1 *.py | wc -l)
-            if [ "$cnt" -eq "1" ]; then
-                APP_FILE=$(cd $APP_ROOT/src/; ls *.py)
-            else
-                echo "Error, no APP_FILE set and $cnt py file(s) found"
-                app_exit
-            fi
+    if [ -z "$APP_FILE"]; then
+        if [ -n "$APP_LANG" ]; then
+            case "$APP_LANG" in
+                java | scala)
+                    file_count "*.jar"
+                    ;;
+                python)
+                    file_count "*.py"
+                    ;;
+                r)
+                    file_count "*.R"
+		    ;;
+                *)
+                    echo "Unrecognized value '$APP_LANG' for APP_LANG and APP_FILE not set"
+                    app_exit
+                    ;;
+            esac
         fi
```

Comment:
probably should have a hard error here, APP_LANG should be set at image build time so its absence is a programmer error

Reply:
this is as easy as removing the  if [ -n "$APP_LANG" ] test before the case